binderhub/builder.py

```diff
@@ -312,11 +312,50 @@ def escape(s):
         if not failed:
             BUILD_TIME.labels(status='success').observe(time.perf_counter() - build_starttime)
             with LAUNCHES_INPROGRESS.track_inprogress():
-                await self.launch()
+                await self.launch(kube)
 
 
-    async def launch(self):
+    async def launch(self, kube):
         """Ask JupyterHub to launch the image."""
+        # check quota first
+        quota = self.settings.get('per_repo_quota')
+
+        image_no_tag = self.image_name.rsplit(':', 1)[0]
+        matching_pods = 0
+        total_pods = 0
+        pool = self.settings['build_pool']
+        f = pool.submit(kube.list_namespaced_pod,
```

Comment:
Longer term this should be a watch rather than making an individual API request for each launch, but is ok for now!